qemu/tests/blockdev_inc_backup_xpt_incon_bitmap.py

```diff
@@ -0,0 +1,159 @@
+import json
+import logging
+import time
+import string
+from avocado.utils import process
+
+from provider import backup_utils
+from provider.blockdev_base import BlockdevBaseTest
+from provider.nbd_image_export import QemuNBDExportImage
+
+from virttest import utils_misc
+
+
+class BlkdevIncXptInconBitmap(BlockdevBaseTest):
+
+    def __init__(self, test, params, env):
+        super(BlkdevIncXptInconBitmap, self).__init__(test, params, env)
+        self.source_images = []
+        self.full_backups = []
+        self.bitmaps = []
+        self.nbd_exports = []
+        self.src_img_tags = params.objects("source_images")
+        list(map(self._init_arguments_by_params, self.src_img_tags))
+
+    def _init_arguments_by_params(self, tag):
+        image_params = self.params.object_params(tag)
+        image_chain = image_params.objects("image_backup_chain")
+        self.source_images.append("drive_%s" % tag)
+        self.full_backups.append("drive_%s" % image_chain[0])
+        self.bitmaps.append("bitmap_%s" % tag)
+        image_params["nbd_export_bitmaps"] = "bitmap_%s" % tag
+        self.nbd_exports.append(QemuNBDExportImage(image_params, tag))
+
+    def do_full_backup(self):
+        extra_options = {"sync": "full",
+                         "persistent": True,
+                         "auto_disable_bitmap": False}
+        backup_utils.blockdev_batch_backup(
+            self.main_vm,
+            self.source_images,
+            self.full_backups,
+            list(self.bitmaps),
+            **extra_options)
+
+    def prepare_data_disk(self, tag):
+        """
+        Override this function, only make fs and mount it
+        :param tag: image tag
+        """
+        self.format_data_disk(tag)
+
+    def gen_inc_files(self):
+        return list(map(self.generate_data_file, self.src_img_tags))
+
+    def kill_vm_after_restart(self):
+        logging.info("Re-start vm again")
+        self.main_vm.create()
+        time.sleep(20)
+        logging.info("Kill vm after its start")
+        pid = self.main_vm.process.get_pid()
+        logging.debug("Ending VM %s process (killing PID %s)",
+                      self.main_vm.name, pid)
+        try:
+            utils_misc.kill_process_tree(pid, 9, timeout=60)
+            logging.debug("VM %s down (process killed)", self.main_vm.name)
+        except RuntimeError:
+            # If all else fails, we've got a zombie...
+            logging.error("VM %s (PID %s) is a zombie!", self.man_vm.name,
+                          self.main_vm.process.get_pid())
+        finally:
+            self.main_vm._cleanup(free_mac_addresses=True)
```

Comment:
What about:
self.main_vm.monitors = []
self.main_vm.destroy(gracefully=False)
So we can avoid a direct call to function _cleanup, which is not public.

Reply:
In qemu_vm.py:
def destroy():
     if gracefully:
                self.graceful_shutdown(kill_timeout)
     if self.monitor:  
                # Try to finish process with a monitor command
                try:
                    self.monitor.quit()
  ...

If we use  self.main_vm.destroy(gracefully=False),  self.monitor.quit() will be called, which is not as expected.
So if we want to call self.main_vm.destroy(gracefully=False), we need to set self.monitor = None at first.
What's your opinion? @zhencliu

Reply:
That's why we set self.main_vm.monitors = [] first, then self.monitor is None, self.monitor is a getter, you can refer to the vt code for details

Reply:
Have updated, thanks zhenchao.